src/game/AI/ScriptDevAI/system/system.cpp

```diff
@@ -81,13 +81,22 @@ void SystemMgr::LoadScriptWaypoints()
     {
         BarGoLink bar(result->GetRowCount());
         uint32 nodeCount = 0;
+        std::set<uint32> blacklistWaypoints;
 
         do
         {
             bar.step();
             Field* fields = result->Fetch();
 
-            uint32 entry  = fields[0].GetUInt32();
+            uint32 entry = fields[0].GetUInt32();
+            uint32 pathId = fields[1].GetUInt32();
+            uint32 pointId = fields[2].GetUInt32();
+
+            if (pointId == 0)
+            {
+                blacklistWaypoints.insert((entry << 8) + pathId);
+                error_db_log("SD2: DB table `script_waypoint` has invalid point 0 for entry %u in path %u.", entry, pathId);
```

Comment:
As it seems we are going the route of sticking to "point index should not be zero", I think at least this error log should be added in core to help stupid DB devs like me to spot errors in their data. Thus, waypoints with first point having index=0 should be reported and skipped, like we usually do with invalid data at loading.

This is for silly DB devs to ensure their data are correct, not for the core to assume they are wrong and try to guess-fix them.

The main real issue here is that we have long-time unwritten (and as such forgotten) convention that led to the current situation. I have then written it in the wiki so at least, this one should not happen anymore.